integration-tester/tester/logic.go

```diff
@@ -0,0 +1,255 @@
+								// These parameters get recorded in Avro messages but otherwise
+								// do not affect any system behavior, so the values don't matter.
+								"--batch-start-time", "1000000000",
+								"--batch-end-time", "1000000100",
+								"--dimension", "123",
+								"--epsilon", "0.23",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// ReadDataShareProcessorSpecificManifest retrieves a manifest.DataShareProcessorSpecificManifest from the given url
+func ReadDataShareProcessorSpecificManifest(url string) (manifest.DataShareProcessorSpecificManifest, error) {
+	manifest := manifest.DataShareProcessorSpecificManifest{}
+	client := http.Client{Timeout: 10 * time.Second}
+
+	r, err := client.Get(url)
+	if err != nil {
+		return manifest, fmt.Errorf("unable to get %s: %v", url, err)
+	}
+	defer r.Body.Close()
+
+	err = json.NewDecoder(r.Body).Decode(&manifest)
+	if err != nil {
+		return manifest, fmt.Errorf("unable to decode body %s: %v", r.Body, err)
+	}
+	return manifest, err
+}
+
+// ReadDataShareProcessorSpecificManifest retrieves a manifest.IngestorGlobalManifest from the given url
+func ReadIngestorGlobalManifests(url string) (manifest.IngestorGlobalManifest, error) {
+	m := manifest.IngestorGlobalManifest{}
+	client := http.Client{Timeout: 10 * time.Second}
+
+	r, err := client.Get(url)
+	if err != nil {
+		return m, fmt.Errorf("unable to get %s: %v", url, err)
+	}
+	defer r.Body.Close()
+
+	err = json.NewDecoder(r.Body).Decode(&m)
+	if err != nil {
+		return m, fmt.Errorf("unable to decode body %s: %v", r.Body, err)
+	}
+	return m, err
+}
+
+func (t *Tester) getValidPacketDecryptionKey(manifest manifest.DataShareProcessorSpecificManifest) (string, error) {
+	for _, value := range manifest.PacketEncryptionKeyCSRs {
+		publicKey, err := getBase64PublicKeyFromCSR(value.CertificateSigningRequest)
+		if err != nil {
+			return "", fmt.Errorf("error when parsing the packet encryption csr: %v", err)
+		}
+
+		return publicKey, nil
+	}
+	return "", fmt.Errorf("no packet encryption CSRs in manifest: %v", manifest)
+}
+
+func getBase64PublicKeyFromCSR(pemCsr string) (string, error) {
+	pemBlock, rest := pem.Decode([]byte(pemCsr))
+	if len(rest) != 0 {
+		return "", fmt.Errorf("unable to pem.Decode the CSR")
+	}
+
+	if pemBlock.Type != "CERTIFICATE REQUEST" {
+		return "", fmt.Errorf("csr not a certificate request")
+	}
+
+	csr, err := x509.ParseCertificateRequest(pemBlock.Bytes)
+	if err != nil {
+		return "", fmt.Errorf("error when parsing the certifcate request: %v", err)
+	}
+	ecdsaPublicKey, ok := csr.PublicKey.(*ecdsa.PublicKey)
+
+	if !ok {
+		return "", fmt.Errorf("certificate request public key was not an ecdsa.PublicKey")
+	}
+	publicKeyBytes := elliptic.Marshal(ecdsaPublicKey.Curve, ecdsaPublicKey.X, ecdsaPublicKey.Y)
+
+	return base64.StdEncoding.EncodeToString(publicKeyBytes), nil
+}
+
+func (t *Tester) getValidBatchSigningKey(manifest manifest.IngestorGlobalManifest) (*corev1.Secret, error) {
```

Comment:
Suggest renaming this type to `IngestorGlobal` so that then it reads as `manifest.IngestorGlobal`.